### PR TITLE
[CES-634] initramfs builtin drivers

### PR DESCRIPTION
**Pull request by marijnd**
A booting kernel with an init.sh that can mount over nfs and set the BOM number

### CODE REVIEW

File: dts/imx8mm-cgtsx8m-ultimain5.0-lvds-1024x600.dts [hunk 1]
```diff
@@ -92,7 +91,30 @@
...
+};
+
+/* Ultimainboard J8 debug console. We cannot use DMA, it uses proprietary firmware drivers.  */
+&uart4 { /* SER1 */
+    /delete-property/ dmas;
+    /delete-property/ dma-names;
+};
+
 / {
+    chosen { /* Investigate how this works, because this is usefull */
```
Comment (raysiudak) [inline dts/imx8mm-cgtsx8m-ultimain5.0-lvds-1024x600.dts:113]: this block can be removed

File: initramfs/init.sh [hunk 2]
```diff
@@ -139,8 +136,13 @@ critical_error()
 
 boot_root()
 {
-    echo "Mounting ${root}."
-    mount -t "${rootfstype}" -o exec,suid,dev,noatime,"${rootflags},${rwmode}" "${root}" "${ROOT_MOUNT}"
+    root_device=${root}
```
Comment (raysiudak) [inline initramfs/init.sh:139]: Should add quotes root_device="${root}", Shellcheck will complain about this, this prevents globbing.

File: initramfs/init.sh [hunk 3]
```diff
@@ -139,8 +136,13 @@ critical_error()
...
+    if [ "${rootfstype}" = "nfs" ]
+    then
+        root_device=${nfs_root}
+    fi
+    echo "Mounting ${root_device}."
```
Comment (raysiudak) [inline initramfs/init.sh:144]: Same here

File: initramfs/init.sh [hunk 4]
```diff
@@ -153,67 +155,23 @@ boot_root()
         restart
     fi
 
-    echo "Starting linux on ${root} of type ${rootfstype} with init=${init}."
+    echo "Starting linux on ${nfs_root} of type ${rootfstype} with init=${init}."
```
Comment (raysiudak) [inline initramfs/init.sh:158]: This should be root_device

**Review by raysiudak — CHANGES_REQUESTED** [review]

File: dts/imx8mm-cgtsx8m-ultimain5.0-lvds-1024x600.dts (hunk 1 above)
Comment (marijnd) [reply dts/imx8mm-cgtsx8m-ultimain5.0-lvds-1024x600.dts:113]: Which part do you want removed? Just the comment, the 'chosen' block or the whole '/' block?

Comment (raysiudak) [reply dts/imx8mm-cgtsx8m-ultimain5.0-lvds-1024x600.dts:113]: The whole chosen block.

**Review by raysiudak — APPROVED** [review]
Looks good, before you merge, can you put it in component test. I can then merge Tom's changes in this branch and test it as a whole.

**Pull request merged**